File: dashboard/src/types/conversation.ts
export enum InterestLevel {
  Low = 1,
  Medium = 2,
  High = 3,
  VeryHigh = 4
}

export enum Timing {
  Immediate = 1,
  ThisWeek = 2,
  ThisMonth = 3,
  NextMonth = 4,
  ThreeMonths = 5,
  Undefined = 6
}

export enum ConversationStatus {
  Active = 1,
  WaitingResponse = 2,
  ProposalSent = 3,
  Closed = 4,
  Lost = 5
}

export interface Conversation {
  id: string;
  leadId: string;
  leadName: string;
  interestLevel: InterestLevel;
  timing: Timing;
  notes: string;
  nextStep?: string | null;
  status: ConversationStatus;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateConversationRequest {
  leadId: string;
  interestLevel: InterestLevel;
  timing: Timing;
  notes: string;
  nextStep?: string;
}

export interface UpdateConversationRequest {
  interestLevel: InterestLevel;
  timing: Timing;
  notes: string;
  nextStep?: string;
  status: ConversationStatus;
}

File: backend/Codebymister.Domain/Entities/Conversation.cs
using Codebymister.Domain.Common;
using Codebymister.Domain.Enums;

namespace Codebymister.Domain.Entities;

public class Conversation : BaseEntity
{
    public Guid LeadId { get; private set; }
    public InterestLevel InterestLevel { get; private set; }
    public Timing Timing { get; private set; }
    public string Notes { get; private set; }
    public string? NextStep { get; private set; }
    public ConversationStatus Status { get; private set; }

    public Lead Lead { get; private set; }

    private Conversation() { }

    public Conversation(
        Guid leadId,
        InterestLevel interestLevel,
        Timing timing,
        string notes,
        string? nextStep = null)
    {
        LeadId = leadId;
        InterestLevel = interestLevel;
        Timing = timing;
        Notes = notes;
        NextStep = nextStep;
        Status = ConversationStatus.Active;
    }

    public void Update(
        InterestLevel interestLevel,
        Timing timing,
        string notes,
        string? nextStep,
        ConversationStatus status)
    {
        InterestLevel = interestLevel;
        Timing = timing;
        Notes = notes;
        NextStep = nextStep;
        Status = status;
    }

    public void UpdateStatus(ConversationStatus status)
    {
        Status = status;
    }
}
